website/docs/src/content/cloud/storage/blob-storage.mdx

```diff
@@ -0,0 +1,287 @@
+---
+title: Blob storage
+description: Store and retrieve unstructured data with GenSX blob storage
+---
+
+# Blob storage
+
+Blob storage provides zero-configuration persistent storage for your GenSX applications. It enables you to store JSON, text, or binary data and per-agent or per-workflow storage without worrying about managing infrastructure.
+
+## Basic usage
+
+To use blob storage in your GenSX application:
+
+1. Install the storage package:
+
+   ```bash
+   npm install @gensx/storage
+   ```
+
+2. Add the `BlobProvider` to your workflow:
+
+   ```tsx
+   import { BlobProvider } from "@gensx/storage";
+
+   const Workflow = ({ input }) => (
+     <BlobProvider>
+       <YourComponent input={input} />
+     </BlobProvider>
+   );
+   ```
+
+3. Access blobs within your components using the `useBlob` hook:
+
+   ```tsx
+   import { useBlob } from "@gensx/storage";
+
+   const blob = useBlob("your-key.json");
+   ```
+
+### Reading blobs
+
+The `useBlob` hook provides simple methods to read different types of data:
+
+```tsx
+import * as gensx from "@gensx/core";
+import { useBlob } from "@gensx/storage";
+
+const ReadData = gensx.Component("ReadData", async ({ key }) => {
+  // Get a reference to a blob
+  const blob = useBlob(`data/${key}.json`);
+
+  // Check if the blob exists
+  const exists = await blob.exists();
+  if (!exists) {
+    return { found: false };
+  }
+
+  // Read JSON data (returns null if doesn't exist)
+  const jsonData = await blob.getJSON();
+
+  // Read as string
+  const textData = await blob.getString();
+
+  // Read as binary with metadata
+  const binaryData = await blob.getRaw();
+
+  return {
+    found: true,
+    jsonData,
+    textLength: textData?.length,
+    contentType: binaryData?.contentType,
+  };
+});
+```
+
+### Writing blobs
+
+You can write data in various formats:
+
+```tsx
+import * as gensx from "@gensx/core";
+import { useBlob } from "@gensx/storage";
+
+const WriteData = gensx.Component(
+  "WriteData",
+  async ({ key, data, format = "json" }) => {
+    // Get a reference to a blob
+    const blob = useBlob(`data/${key}`);
```

Comment:
include .json above but not here so the ReadData and WriteData components would be incompatible